core/domain/exp_domain.py

```diff
@@ -257,15 +256,17 @@ def __init__(
 
     def to_dict(self):
         """Returns a dict representing this ExplorationCommitLogEntry domain
-        object. This omits created_on, user_id and commit_cmds.
+        object. This omits created_on, user_id and commit_cmds and adds username
+        (derived from user_id).
 
         Returns:
             dict. A dict, mapping all fields of ExplorationCommitLogEntry
-            instance, except created_on, user_id and commit_cmds fields.
+            instance, except created_on, user_id and commit_cmds fields and
+            adding username (derived from user_id).
         """
         return {
             'last_updated': utils.get_time_in_millisecs(self.last_updated),
-            'username': self.username,
+            'username': user_services.get_username(self.user_id),
```

Comment:
If we are getting many of these to display in a page, then I think you need to do a get_multi here or it won't scale.

Also, typically, exp_domain shouldn't call services. I think maybe what you want to do is do the replacement in the controller layer in the specific handler, as close to the client/server endpoint as possible.

Reply:
Done.